shared/modules/bus_module.go

```diff
@@ -23,5 +24,6 @@ type Bus interface {
 	GetTelemetryModule() TelemetryModule
 
 	// Configuration
-	GetConfig() *Config
+	GetConfig() map[string]json.RawMessage
```

Comment:
A custom type may be a better choice for a function that returns `Config`. Is there a specific reason for not modifying the `Config` type from the above line that was deleted, rather than committing to a map with specific key/value types?

Reply:
Yes, the idea is to use as many non-custom structures from the shared package as possible in order to prevent cyclical dependencies and minimize the shared module.

A very similar approach is taken in V0 and the Cosmos SDK - i.e each module controls its own `configuration object` which is the proper encapsulation IMO

Reply:
Discussed offline. See other comments.